app/src/main/java/com/example/drawingactivity/DrawActivity.kt

```diff
@@ -0,0 +1,150 @@
+package com.example.drawingactivity
+
+import android.os.Bundle
+import android.widget.Button
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+
+class DrawActivity : AppCompatActivity() {
+
+    private val viewModel: DrawViewModel by viewModels()
+    private val dialogUtils = DialogUtils
+    private var isAddDialogOpen = false
+    private var isEditDialogOpen = false
+    private var isLayerListViewSheetOpen = false
+    private var layerList = mutableListOf<String>()
+    private var currentBottomSheetDialog: ((Int, String?, String) -> Unit)? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val drawableArea = findViewById<DrawableAreaView>(R.id.drawableArea)
+        drawableArea.onBitmapDrawn = { bitmap -> viewModel.setCurrentBitmapState(bitmap) }
+        viewModel.currentBitmap?.let { drawableArea.setBitmap(it) }
+        val colorSelector = findViewById<PrimaryColorCircleView>(R.id.colorCircle)
+        colorSelector.onChangeColor = { viewModel.cycleCurrentColor() }
+
+        if (viewModel.getCurrentLayerList().isEmpty()) {
+            viewModel.addLayer(this.getString(R.string.layerHint, 1))
+        }
+
+        findViewById<Button>(R.id.addLayer).setOnClickListener {
+            viewModel.setDialogState()
+        }
+
+        findViewById<Button>(R.id.viewLayers).setOnClickListener {
+            viewModel.setLayerListViewState()
+        }
+
+        viewModel.apply {
+            onAddLayerComplete = {
+                Toast.makeText(
+                    this@DrawActivity,
+                    this@DrawActivity.getString(R.string.addLayerSuccess, it),
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+            onDeleteLayerComplete = {
+                Toast.makeText(
+                    this@DrawActivity,
+                    this@DrawActivity.getString(R.string.deleteLayerSuccess, it),
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+            onEditLayerComplete = { oldName, newName ->
+                Toast.makeText(
+                    this@DrawActivity,
+                    this@DrawActivity.getString(R.string.editLayerSuccess, oldName, newName),
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+        }
+
+        viewModel.onUpdate = {
+            colorSelector.circlePrimaryColor = it.primaryColor
+            drawableArea.pathPaint.color = it.primaryColor
+            drawableArea.userDrawnBitmap = it.currentBitmap
+            layerList = it.layerList
```

Comment:
I understand holding the state of if dialogs are open, but this isn't needed and seems like a bit of a bad practice. You could get out of sync with the viewmodel and it'd be no bueno. No reason to risk it